### PR TITLE
[chore] Prepare release 0.84.0

### PR DESCRIPTION
**Pull request by opentelemetrybot**
The following commands were run to prepare this release:
- make chlog-update VERSION=v0.84.0
- sed -i.bak s/0.83.0/0.84.0/g versions.yaml
- make multimod-prerelease
- make multimod-sync

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: cmd/oteltestbedcol/builder-config.yaml [hunk 1]
```diff
@@ -2,50 +2,50 @@ dist:
   module: github.com/open-telemetry/opentelemetry-collector-contrib/cmd/oteltestbedcol
   name: oteltestbedcol
   description: OpenTelemetry Collector binary for testbed only tests.
-  version: 0.83.0-dev
+  version: 0.84.0-dev
   otelcol_version: 0.82.0
```
Comment (codeboten) [inline cmd/oteltestbedcol/builder-config.yaml:6]: This should be updated to 0.84.0

Comment (codeboten) [inline cmd/oteltestbedcol/builder-config.yaml:6]:
```suggestion
  otelcol_version: 0.84.0
```

File: cmd/otelcontribcol/builder-config.yaml [hunk 2]
```diff
@@ -2,206 +2,206 @@ dist:
   module: github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol
   name: otelcontribcol
   description: Local OpenTelemetry Collector Contrib binary, testing only.
-  version: 0.83.0-dev
+  version: 0.84.0-dev
   otelcol_version: 0.82.0
```
Comment (codeboten) [inline cmd/otelcontribcol/builder-config.yaml:6]:
```suggestion
  otelcol_version: 0.84.0
```

File: cmd/oteltestbedcol/builder-config.yaml (hunk 1 above)
Comment (dmitryax) [reply cmd/oteltestbedcol/builder-config.yaml:6]: Thanks for catching it. Updated

**Review by songy23 — APPROVED** [review]

**Pull request merged**